### PR TITLE
combine candidatesFromDB func and fix putpollresultAction issue for easter cut 

### PR DESCRIPTION
**Pull request by koseoyoung**
1. combine candidatesByHeight and candidatesFromDB func 
2. use this new func in CreatePostSystemActions to fix putpollresult action issue in correct way

### CODE REVIEW

File: action/protocol/poll/staking_command.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ type stakingCommand struct {
 	stakingV1   Protocol
 	stakingV2   *staking.Protocol
 	candIndexer *CandidateIndexer
+	sr          protocol.StateReader
```
Comment (yutongp) [inline action/protocol/poll/staking_command.go:30]: I think it is weird to store sr in protocol struct

Comment (CoderZhi) [reply action/protocol/poll/staking_command.go:30]: agree, to have sr in CreatePostSystemActions, you can modify the API to 
`CreatePostSystemActions(ctx context.Context, sr StateReader) ([]action.Envelope, error)`

Comment (koseoyoung) [reply action/protocol/poll/staking_command.go]: address it

File: action/protocol/vote/candidatesutil/candidatesutil.go [hunk 2]
```diff
@@ -41,49 +41,35 @@ const NxtKickoutKey = "NextKickoutKey."
...
-
-// CandidatesFromDB returns array of Candidates at current epoch
-func CandidatesFromDB(sr protocol.StateReader, epochStartPoint bool, opts ...protocol.StateOption) ([]*state.Candidate, uint64, error) {
-	var candidates state.CandidateList
-	candidatesKey := ConstructKey(CurCandidateKey)
-	if epochStartPoint {
-		// if not shifted yet
-		log.L().Debug("Read candidate list with next candidate key")
-		candidatesKey = ConstructKey(NxtCandidateKey)
+	if !beforeEaster {
```
Comment (CoderZhi) [inline action/protocol/vote/candidatesutil/candidatesutil.go:54]: } else {

**Pull request merged**